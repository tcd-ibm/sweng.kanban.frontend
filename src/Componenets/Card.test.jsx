import React from 'react';
import  renderer  from "react-test-renderer";
import { shallow } from 'enzyme';
import KBCard from './Card';



describe('Card Component', () => {

    it("renders without crashing", () => {
        shallow(<KBCard />);
      });
    it('renders a snapshot', () => {
        const cardSnap = renderer.create(<KBCard testid="card"/>).toJSON();
        expect(cardSnap).toMatchSnapshot();
      });

      it('renders the title of the card', () => {
        const cardSnap = renderer.create(<KBCard text="KBCard.title"/>).toJSON();
        expect(cardSnap).toMatchSnapshot();  
       
    });
    it('renders the description in the card', () => {
        const cardSnap = renderer.create(<KBCard text="KBCard.desc"/>).toJSON();
        expect(cardSnap).toMatchSnapshot();  
       
    });

    
});