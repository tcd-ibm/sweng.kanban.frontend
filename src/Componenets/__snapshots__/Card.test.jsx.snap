// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders a snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  testid="card"
>
  <div
    className="MuiBox-root css-cqh5gh"
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        style={
          Object {
            "marginBottom": 12,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Card Component renders the description in the card 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  testid="card"
>
  <div
    className="MuiBox-root css-cqh5gh"
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        style={
          Object {
            "marginBottom": 12,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Card Component renders the title of the card 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  testid="card"
>
  <div
    className="MuiBox-root css-cqh5gh"
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        style={
          Object {
            "marginBottom": 12,
          }
        }
      />
    </div>
  </div>
</div>
`;
